libs/consent/api/src/lib/userConsent/user-consent.service.ts

```diff
@@ -0,0 +1,116 @@
+import {Injectable, NotFoundException} from '@nestjs/common'
+import {PrismaClient} from '@prisma/client'
+import {UserSession} from '@wepublish/authentication/api'
+import {
+  UserConsent,
+  UserConsentInput,
+  UpdateUserConsentInput,
+  UserConsentFilter
+} from './user-consent.model'
+
+@Injectable()
+export class UserConsentService {
+  constructor(private prisma: PrismaClient) {}
+
+  /*
+  Queries
+ */
+  async userConsentList(filter?: UserConsentFilter): Promise<UserConsent[]> {
+    const data = await this.prisma.userConsent.findMany({
+      where: {
+        consent: {
+          ...filter
+        }
+      },
+      orderBy: {
+        createdAt: 'desc'
+      },
+      include: {
+        user: true,
+        consent: true
+      }
+    })
+    return data
+  }
+
+  async userConsent(id: string): Promise<UserConsent> {
+    const data = await this.prisma.userConsent.findUnique({
+      where: {
+        id
+      },
+      include: {
+        user: true,
+        consent: true
+      }
+    })
+
+    if (!data) {
+      throw new NotFoundException(`UserConsent with id ${id} not found`)
+    }
+
+    return data
+  }
+
+  /*
+  Mutations
+ */
+  async createUserConsent(userConsent: UserConsentInput, user: UserSession): Promise<UserConsent> {
+    // only allow creating for admin or affected user
+    if (!user.user.roleIDs.includes('admin') && user.user.id !== userConsent.userId) {
```

Comment:
The service shouldn't be checking permissions but the resolver via decorators, this allows for things to be reused